code/datums/ammo/bullet/sniper.dm

```diff
@@ -107,19 +107,78 @@
 	damage = 125
 	shell_speed = AMMO_SPEED_TIER_6
 
-/datum/ammo/bullet/sniper/anti_materiel/on_hit_mob(mob/M,obj/projectile/P)
-	if((P.projectile_flags & PROJECTILE_BULLSEYE) && M == P.original)
-		var/mob/living/L = M
-		var/size_damage_mod = 0.8
-		if(isxeno(M))
-			var/mob/living/carbon/xenomorph/target = M
-			if(target.mob_size >= MOB_SIZE_XENO)
-				size_damage_mod += 0.6
-			if(target.mob_size >= MOB_SIZE_BIG)
-				size_damage_mod += 0.6
-		L.apply_armoured_damage(damage*size_damage_mod, ARMOR_BULLET, BRUTE, null, penetration)
-		// 180% damage to all targets (225), 240% (300) against non-Runner xenos, and 300% against Big xenos (375). -Kaga
-		to_chat(P.firer, SPAN_WARNING("Bullseye!"))
+/datum/ammo/bullet/sniper/anti_materiel/on_hit_mob(mob/target_mob,obj/projectile/aimed_projectile)
+	if((aimed_projectile.projectile_flags & PROJECTILE_BULLSEYE) && target_mob == aimed_projectile.original)
+
+		var/amr_counter = 0
+
+		if(istype(aimed_projectile.shot_from, /obj/item/weapon/gun/rifle/sniper/XM43E1))
+			var/obj/item/weapon/gun/rifle/sniper/XM43E1/amr = aimed_projectile.shot_from
+			if(target_mob == (amr.focused_fire_target?.resolve()))
+				if(amr.focused_fire_counter < 2)
+					amr.focused_fire_counter += 1
+				else
+					amr.focused_fire_counter = 2
+			else
+				amr.focused_fire_counter = 0
+
+			amr_counter = amr.focused_fire_counter + 1
+			amr.focused_fire_target = WEAKREF(target_mob)
+
+		var/mob/living/living_target = target_mob
+		var/size_damage_mod = 0.8 // 1.8x vs Non-Xenos (225)
```

Comment:
comment and code itself contradict each other

Also, `size_damage_mod` goes to 0 when shot at T3 xenos, due -0.2 on successful xeno check and -0.6 on `MOB_SIZE_BIG` check.
So `((damage * size_damage_mod) + ((target.health + damage) * size_current_health_damage))` becomes just `(target.health + damage) * size_current_health_damage)` capping damage to `(target.health + 125) * 0.3`